Demo/Demo/ViewModels/BaseViewModel.swift

```diff
@@ -0,0 +1,102 @@
+    }
+
+    // MARK: - Helper Functions
+
+    func updateTitle(_ message: String) {
+        DispatchQueue.main.async {
+            self.view?.bottomStatusLabel.text = message
+        }
+    }
+
+    func processOrder(orderID: String, completion: @escaping () -> Void = {}) {
+        switch DemoSettings.intent {
+        case .authorize:
+            updateTitle("Authorizing order...")
+        case .capture:
+            updateTitle("Capturing order...")
+        }
+
+        let processOrderParams = ProcessOrderParams(
+            orderId: orderID,
+            intent: DemoSettings.intent.rawValue,
+            countryCode: "US"
+        )
+
+        DemoMerchantAPI.sharedService.processOrder(processOrderParams: processOrderParams) { result in
+            switch result {
+            case .success(let order):
+                self.updateTitle("\(DemoSettings.intent.rawValue.capitalized) success: \(order.status)")
+                print("✅ processed orderID: \(order.id) with status: \(order.status)")
+            case .failure(let error):
+                self.updateTitle("\(DemoSettings.intent.rawValue.capitalized) fail: \(error.localizedDescription)")
+                print("❌ failed to process orderID: \(error)")
+            }
+            completion()
+        }
+    }
+
+    func createCard(cardNumber: String?, expirationDate: String?, cvv: String?) -> Card? {
+        guard let cardNumber = cardNumber, let expirationDate = expirationDate, let cvv = cvv else {
+            updateTitle("Failed: missing card / orderID.")
+            return nil
+        }
+
+        let cleanedCardText = cardNumber.replacingOccurrences(of: " ", with: "")
+
+        let expirationComponents = expirationDate.components(separatedBy: " / ")
+        let expirationMonth = expirationComponents[0]
+        let expirationYear = "20" + expirationComponents[1]
+
+        return Card(number: cleanedCardText, expirationMonth: expirationMonth, expirationYear: expirationYear, securityCode: cvv)
+    }
+
+    func checkoutWithCard(_ card: Card, orderID: String, completion: @escaping () -> Void = { }) {
+        let config = CoreConfig(clientID: DemoSettings.clientID, environment: DemoSettings.environment.paypalSDKEnvironment)
+        let cardClient = CardClient(config: config)
+        let cardRequest = CardRequest(orderID: orderID, card: card)
+
+        cardClient.approveOrder(request: cardRequest) { result in
+            switch result {
+            case .success(let result):
+                self.updateTitle("\(DemoSettings.intent.rawValue.capitalized) status: APPROVED")
+                self.processOrder(orderID: result.orderID) {
+                    completion()
+                }
+            case .failure(let error):
+                self.updateTitle("\(DemoSettings.intent) failed: \(error.localizedDescription)")
+            }
+        }
+    }
+
+    func enablePayButton(cardNumber: String, expirationDate: String, cvv: String) -> Bool {
```

Comment:
I think this might make more sense as `isCardFormValid` or `validateCardFields` or something like that since it's not actually enabling the button.

Reply:
I think `isCardFormValid` makes this way more clear - thanks for this suggestion!

Reply:
I think `isCardFormValid` makes this way more clear - thanks for this suggestion!